src/main/kotlin/org/jetbrains/research/testspark/tools/llm/test/JUnitTestSuitePresenter.kt

```diff
@@ -57,16 +73,29 @@ class JUnitTestSuitePresenter(
         testCaseIndex: Int,
     ): String =
         testSuite.run {
-            JavaClassBuilderHelper.generateCode(
-                project,
-                getClassWithTestCaseName(testCases[testCaseIndex].name),
-                testCases[testCaseIndex].toStringWithoutExpectedException() + "\n",
-                imports,
-                packageString,
-                runWith,
-                otherInfo,
-                generatedTestsData,
-            )
+            when (language) {
+                Language.Java -> JavaClassBuilderHelper.generateCode(
+                    project,
+                    getClassWithTestCaseName(testCases[testCaseIndex].name),
+                    testCases[testCaseIndex].toStringWithoutExpectedException() + "\n",
+                    imports,
+                    packageName,
+                    runWith,
+                    otherInfo,
+                    generatedTestsData,
+                )
+
+                Language.Kotlin -> KotlinClassBuilderHelper.generateCode(
+                    project,
+                    getClassWithTestCaseName(testCases[testCaseIndex].name),
+                    testCases[testCaseIndex].toStringWithoutExpectedException() + "\n",
+                    imports,
+                    packageName,
+                    runWith,
+                    otherInfo,
+                    generatedTestsData,
+                )
+            }
```

Comment:
This consturct could also be replaced with a single function call if `generateCode` was non-static.

Reply:
https://github.com/JetBrains-Research/TestSpark/pull/282#discussion_r1686651574